### ISSUE TITLE
Use the top-level parameter object_lock_enabled and the aws_s3_bucket_object_lock_configuration resource instead

### ISSUE THREAD

**Issue by dnyaneshwar-sonawane** [terraform-aws-modules/terraform-aws-s3-bucket#148]
## Description

```
Warning: Argument is deprecated

  on ../modules/aws/s3-bucket/main.tf line 12, in resource "aws_s3_bucket" "this":
  12: resource "aws_s3_bucket" "this" {

Use the top-level parameter object_lock_enabled and the
aws_s3_bucket_object_lock_configuration resource instead

(and 15 more similar warnings elsewhere)
```

Getting this error after upgrading was the provider version. 

## Versions

- Module version [Required]:

- Terraform version:
```
Terraform v0.13.1
+ provider registry.terraform.io/hashicorp/aws v4.9.0
```

**Comment by antonbabenko** [comment]
This issue has been resolved in version 3.1.0 :tada:

**Comment by antonbabenko** [comment]
Please take a look at the example code in #149
